### ISSUE TITLE
periodicity 'dekadal'?

### ISSUE THREAD

**Issue by rabons** [monocongo/climate_indices#554]
**Is your feature request related to a problem? Please describe.**
I would like to calculate SPI values on dekadal periodicity, i.e. every ten days of a month (days 1-10 are dekad 1, days 11-20 dekad 2, days >= 21 are dekad 3 of each month). This is often used in agricultural drought assessments.

**Describe the solution you'd like**
Ideally, I would like to be able to run the spi command with `--periodicity dekad`, where the input netCDF file for precipitation is equally in dekadal time steps

**Describe alternatives you've considered**
I may have to consider using a different package altogether.

**Comment by monocongo** [comment]
Yes, this should be possible. We initially started with only monthly data but later added daily. So this would continue that work to add a new periodicity.

**Comment by rabons** [comment]
Thank you! Could you briefly point me to the files which would require modifying? I could attempt at doing it myself as a starter.

**Comment by monocongo** [comment]
Can anyone comment as to strategies we might use for testing our implementation of this new period type?

**Comment by monocongo** [comment]
Thanks @rabons 

We need to add a new enumeration to the class `compute.Periodicity`, then make sure we can support 36 periods per year for each index type. Initially we can look for places where the check for monthly/daily is happening then add another case for dekadal datasets where it makes a difference.

**Comment by bennyistanto** [comment]
> Can anyone comment as to strategies we might use for testing our implementation of this new period type?

+1 for this `--periodicity dekad` feature.

@rabons I just completed test using global dekad CHIRPS version 3.0, and calculate monthly accumulation rolling by dekad. Then calculate the SPI separately for dekad1, 2 and 3. Each of them produce 71 GB of file output for single scale (I tested for scale 6-month).

Concatenate all three together and transpose to `time,lat,lon` and save with compression using xarray, I got 57 GB for the final output (Global dekad SPI-6, 1981-2024, 0.05 spatial resolution).

![Image](https://github.com/user-attachments/assets/a549f7b8-4531-4c47-94b4-8f01abe910ad)
